src/Google/GlobalSiteTag.php

```diff
@@ -758,6 +767,12 @@ private function normalize_and_hash( $value, $algo = 'sha256' ): string {
 	 * @return bool
 	 */
 	private function is_gtg_enabled(): bool {
-		return (bool) $this->options->get( OptionsInterface::ADS_GTG_ENABLED );
+		// Disable GTG if the feature flag has marked it as disabled.
+		if ( ! $this->features->is_enabled( Features::GOOGLE_TAG_GATEWAY ) ) {
```

Comment:
This will call WCS every time the website loads, even when using the cached version. Since the transient expires every 5 minutes, is it okay to send requests this frequently? High-traffic sites may end up making many calls to WCS.

```
$this->transients->set( $this->transient_name(), $option, MINUTE_IN_SECONDS * 5 );
```